### PR TITLE
Upgrade from node 20 to node 22

### LINKED ISSUE #7724: Upgrade to node 22
### Summary
Upgrade to node 22, which was released April 2024.

### Motivation and timeline
Node 20 won't be out of LTS anytime soon so we have a while, but making this for when we'll eventually need to upgrade.

Related dependabot PR for our frontend images: https://github.com/CDCgov/prime-simplereport/pull/7698

### Steps
Previous upgrade for reference: https://github.com/CDCgov/prime-simplereport/pull/7549
For changes, see: https://nodejs.org/en/blog/announcements/v22-release-announce

Upgrade node to version 22 in the following places:

- [x] SimpleReport app - `frontend`
- [x] SimpleReport app - `cypress`
- [x] SimpleReport static site
- [ ] Azure function app - reference PR from node 20 upgrade [here](https://github.com/CDCgov/prime-simplereport/pull/7595)
- [x] Docker images
- [x] Deploy jobs / actions (if any are hard pinned)
- [ ] Perform bug bash (see previous structure / flows [here](https://docs.google.com/document/d/16cfE6gsslhcgvH3-Gi9LfkL85FQ-hEMPBkpLzR9dWWw/edit?usp=sharing))

### PR DESCRIPTION
**Pull request by bobbywells52**
# DEVOPS PULL REQUEST

## Related Issue

- Resolves #7724 

## Changes Proposed

Upgrade from node 20 to node 22 in the following places:
- SimpleReport app - `frontend`
- SimpleReport app - `cypress`
- SimpleReport static site ([see static site PR here](https://github.com/CDCgov/prime-simplereport-site/pull/712))
- Docker images
- Deploy jobs / actions 

## Additional Information

- Note that the Azure function app has not been upgraded to node 22 as Azure [only supports 20.x versions of node currently](https://learn.microsoft.com/en-us/azure/azure-functions/functions-reference-node?tabs=javascript%2Cwindows%2Cazure-cli&pivots=nodejs-model-v4#supported-versions).

- ~~While working with the dependencies I noticed that we are no longer using: `loader-utils`, `minimist`, `prop-types`, and `browserslist`. Given that any dependency is a potential vulnerability and that we are no longer using them, I removed them from our lock files in this PR as well.~~

## Testing

- Live to be tested in dev6
- Engineering team will complete bug bash together [here](https://docs.google.com/document/d/19EdZErh-P4aK2NbZAiNtBcQHymbBpV5bmk8lKOw0Ras/edit#heading=h.x3o4nsu2ptqi)

---

### CODE REVIEW

File: backend/src/main/resources/application-azure-dev6.yaml [hunk 1]
```diff
@@ -6,7 +6,7 @@ simple-report:
   patient-link-url: https://dev6.simplereport.gov/app/pxp?plid=
   twilio-callback-url: https://dev6.simplereport.gov/api/pxp/callback
   sendgrid:
-    enabled: false
+    enabled: true
```
Comment (emyl3) [inline backend/src/main/resources/application-azure-dev6.yaml:9]: I think this was accidentally added while testing today 😅

**Review by fzhao99 — COMMENTED** [review]
Could I ask how we went about verifying the unused deps? Ran `yarn why` off main for those packages and noticed that some dependencies get listed

`loader-utils`
```
 yarn why loader-utils
yarn why v1.22.22
[1/4] 🤔  Why do we have the module "loader-utils"...?
[2/4] 🚚  Initialising dependency graph...
[3/4] 🔍  Finding dependency...
[4/4] 🚡  Calculating file sizes...
=> Found "loader-utils@3.2.1"
info Has been hoisted to "loader-utils"
info Reasons this module exists
   - Specified in "dependencies"
   - Hoisted from "react-scripts#react-dev-utils#loader-utils"
```

`minimist`

```
yarn why minimist            
yarn why v1.22.22
[1/4] 🤔  Why do we have the module "minimist"...?
[2/4] 🚚  Initialising dependency graph...
[3/4] 🔍  Finding dependency...
[4/4] 🚡  Calculating file sizes...
=> Found "minimist@1.2.8"
info Has been hoisted to "minimist"
info Reasons this module exists
   - Specified in "dependencies"
   - Hoisted from "eslint-plugin-import#tsconfig-paths#minimist"
   - Hoisted from "eslint-plugin-import#tsconfig-paths#json5#minimist"
   - Hoisted from "stylelint#postcss-sass#gonzales-pe#minimist"
   - Hoisted from "react-scripts#tailwindcss#detective#minimist"
   - Hoisted from "@storybook#react-webpack5#@storybook#preset-react-webpack#tsconfig-paths#minimist"
   - Hoisted from "react-scripts#@svgr#webpack#@svgr#plugin-svgo#svgo#mkdirp#minimist"
```
`prop-types`
```
yarn why prop-types        
yarn why v1.22.22
[1/4] 🤔  Why do we have the module "prop-types"...?
[2/4] 🚚  Initialising dependency graph...
[3/4] 🔍  Finding dependency...
[4/4] 🚡  Calculating file sizes...
=> Found "prop-types@15.8.1"
info Has been hoisted to "prop-types"
info Reasons this module exists
   - Specified in "dependencies"
   - Hoisted from "react-transition-group#prop-types"
   - Hoisted from "@apollo#client#prop-types"
   - Hoisted from "@szhsin#react-menu#prop-types"
   - Hoisted from "react-modal#prop-types"
   - Hoisted from "@storybook#react#prop-types"
   - Hoisted from "@fortawesome#react-fontawesome#prop-types"
   - Hoisted from "@types#apollo-upload-client#@apollo#client#prop-types"
   - Hoisted from "react-scripts#eslint-config-react-app#eslint-plugin-react#prop-types"
```

`browserslist` - we're using this in a few css-related contexts, so wondering if this might be responsible for Dan's flag on the PXP page for some visual regressions
```
yarn why browserslist          
yarn why v1.22.22
[1/4] 🤔  Why do we have the module "browserslist"...?
[2/4] 🚚  Initialising dependency graph...
[3/4] 🔍  Finding dependency...
[4/4] 🚡  Calculating file sizes...
=> Found "browserslist@4.21.10"
info Has been hoisted to "browserslist"
info Reasons this module exists
   - Specified in "dependencies"
   - Hoisted from "webpack#browserslist"
   - Hoisted from "react-scripts#browserslist"
   - Hoisted from "stylelint#autoprefixer#browserslist"
   - Hoisted from "react-scripts#react-dev-utils#browserslist"
   - Hoisted from "@babel#helper-define-polyfill-provider#@babel#helper-compilation-targets#browserslist"
   - Hoisted from "react-scripts#postcss-preset-env#browserslist"
   - Hoisted from "@storybook#react-webpack5#@storybook#builder-webpack5#webpack#browserslist"
   - Hoisted from "@storybook#react-webpack5#@storybook#preset-react-webpack#webpack#browserslist"
   - Hoisted from "react-scripts#postcss-preset-env#autoprefixer#browserslist"
   - Hoisted from "react-scripts#css-minimizer-webpack-plugin#cssnano#cssnano-preset-default#postcss-colormin#browserslist"
   - Hoisted from "@graphql-tools#relay-operation-optimizer#@ardatan#relay-compiler#babel-preset-fbjs#@babel#plugin-transform-function-name#@babel#helper-compilation-targets#browserslist"
   - Hoisted from "@graphql-tools#relay-operation-optimizer#@ardatan#relay-compiler#babel-preset-fbjs#@babel#plugin-proposal-object-rest-spread#@babel#helper-compilation-targets#browserslist"
   - Hoisted from "react-scripts#babel-preset-react-app#@babel#plugin-transform-runtime#babel-plugin-polyfill-corejs3#core-js-compat#browserslist"
   - Hoisted from "react-scripts#css-minimizer-webpack-plugin#cssnano#cssnano-preset-default#postcss-convert-values#browserslist"
   - Hoisted from "react-scripts#css-minimizer-webpack-plugin#cssnano#cssnano-preset-default#postcss-merge-rules#browserslist"
   - Hoisted from "react-scripts#css-minimizer-webpack-plugin#cssnano#cssnano-preset-default#postcss-minify-params#browserslist"
   - Hoisted from "react-scripts#css-minimizer-webpack-plugin#cssnano#cssnano-preset-default#postcss-normalize-unicode#browserslist"
   - Hoisted from "react-scripts#css-minimizer-webpack-plugin#cssnano#cssnano-preset-default#postcss-reduce-initial#browserslist"
   - Hoisted from "react-scripts#css-minimizer-webpack-plugin#cssnano#cssnano-preset-default#postcss-colormin#caniuse-api#browserslist"
   - Hoisted from "react-scripts#css-minimizer-webpack-plugin#cssnano#cssnano-preset-default#postcss-merge-longhand#stylehacks#browserslist"
```

**Comment by DanielSass** [comment]
But to Bob's point - the dependencies removed from the package.json are still included in the project via other dependencies (just run a `yarn list` and search for them).

I believe they were listed in the package.json explicitly to force their versions to the most recent package (probably for snyk reasons?).

**Comment by mpbrown** [comment]
> Could I ask how we went about verifying the unused deps? Ran `yarn why` off main for those packages and noticed that some dependencies get listed

Was curious about why those dependencies get listed at the top level in `package.json` if we aren't using them directly. If I'm understanding [hoisting](https://www.jonathancreamer.com/inside-the-pain-of-monorepos-and-hoisting/) correctly, [yarn transforms the dependency tree](https://yarnpkg.com/advanced/lexicon#hoisting) to optimize it by removing as many nodes as possible and figuring out what the most common versions of dependencies are. So I think we'd want to keep these dependencies listed in `package.json` if yarn says it hoisted it from other dependencies.

As a side note, I didn't know yarn version 2 (Yarn Modern) is out and [apparently they recommending upgrading if possible](https://yarnpkg.com/migration/overview)

File: frontend/package.json [hunk 2]
```diff
@@ -22,18 +22,14 @@
...
     "flagged": "^2.0.10",
     "google-libphonenumber": "^3.2.33",
     "graphql": "^16.6.0",
     "i18next": "^23.5.1",
     "jwt-decode": "^3.1.2",
-    "loader-utils": "^3.2.1",
     "lodash": "^4.17.21",
-    "minimist": "1.2.8",
     "moment": "^2.29.4",
-    "prop-types": "^15.7.2",
```
Comment (DanielSass) [inline frontend/package.json:36]: I believe we need to revert this change

**Review by DanielSass — CHANGES_REQUESTED** [review]

File: frontend/package.json (hunk 2 above)
Comment (bobbywells52) [reply frontend/package.json]: @DanielSass reverted and should be good to go 🫡

**Review by DanielSass — APPROVED** [review]
LGTM

**Review by emyl3 — APPROVED** [review]
Nice work, Bobby!

**Review by shanice-skylight — APPROVED** [review]
I added a new patient and ran 3 tests and all went as they should. LGTM.
